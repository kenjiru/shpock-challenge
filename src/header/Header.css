.header {
    background: #45b862;
}

.header .logo {
    height: 60px;
    margin-top: 23px;
    background: #45b862 url("/img/logo.png") no-repeat;
}
