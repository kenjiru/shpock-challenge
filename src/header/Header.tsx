import * as React from "react";
import { Component, ReactElement } from "react";
import { Grid, Row, Col } from "react-flexbox-grid";

import "./Header.css";

class Header extends Component<{}> {
    public render(): ReactElement<HTMLElement> {
        return (
            <div className="header">
                <Grid>
                    <Row>
                        <Col xs={0} md={2} lg={3}/>
                        <Col xs={12} md={8} lg={6}>
                            <div className="logo"/>
                        </Col>
                        <Col xs={0} md={2} lg={3}/>
                    </Row>
                </Grid>
            </div>
        );
    }
}

export default Header;
