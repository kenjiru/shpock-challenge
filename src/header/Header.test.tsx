import * as React from "react";
import { shallow } from "enzyme";
import shallowToJson from "enzyme-to-json";

import Header from "./Header";

describe("Header", () => {
    it("should render correctly", () => {
        const output = shallow(
            <Header/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});
