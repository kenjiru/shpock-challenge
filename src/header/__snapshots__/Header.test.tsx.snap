// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  className="header"
>
  <Grid>
    <Row>
      <Col
        lg={3}
        md={2}
        xs={0}
      />
      <Col
        lg={6}
        md={8}
        xs={12}
      >
        <div
          className="logo"
        />
      </Col>
      <Col
        lg={3}
        md={2}
        xs={0}
      />
    </Row>
  </Grid>
</div>
`;
